specification/common-types/resource-management/v4/customermanagedkeys.json

```diff
@@ -0,0 +1,70 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "4.0",
+    "title": "Common types"
+  },
+  "paths": {},
+  "definitions": {
+    "encryption": {
+      "type": "object",
+      "description": "(Optional) Discouraged to include in resource definition. Only needed where it is possible to disable platform (AKA infrastructure) encryption. Azure SQL TDE is an example of this. Values are enabled and disabled.",
+      "properties": {
+        "infrastructureEncryption": {
+          "type": "string",
+          "enum": [
+            "enabled",
+            "disabled"
+          ],
+          "x-ms-enum": {
+            "name": "InfrastructureEncryption",
+            "modelAsString": true
+          },
+          "description": "Values are enabled and disabled."
+        },
+        "customerManagedKeyEncryption": {
+          "description": "All Customer-managed key encryption properties for the resource.",
+          "$ref": "#/definitions/customerManagedKeyEncryption"
+        }
+      }
+    },
+    "customerManagedKeyEncryption": {
+      "type": "object",
+      "description": "All Customer-managed key encryption properties for the resource.",
+      "properties": {
```

Comment:
@jackrichins Note that I didn't add "federatedClientId" here. We (for example) don't need to support it.
Can you elaborate where exactly is it needed? is it only for cross cloud?

Reply:
It's for cross tenant - for example Customer X builds a service on Azure. Customer X has a Customer Y that needs CMK with a key in Customer Y's tenant. federatedClientId is needed to support Customer X using Azure encryption at rest but specifying that Customer Y's tenant be used.

Reply:
I see. So as this is meant to be a common-type, would you expect federatedClientId to be included under keyEncryptionKeyIdentity as suggested in the preferred API schema?